### ISSUE TITLE
ACM Fleet Membership GKE Cluster name length issue.

### ISSUE THREAD

**Issue by grsiepka** [terraform-google-modules/terraform-google-kubernetes-engine#1423]
### TL;DR

Long GKE cluster names result in a failure in GKE Hub membership via terraform.  
Example:
36 character cluster name:
17 character Project ID
11 character Region (us-central1)

Total 64 characters

It seems like maybe the fleet membership registration is combining all of the above as the cluster name? 

Terraform run fails with 

| Error: Error creating Membership: googleapi: Error 400: InvalidValueError for field membership_id: must be no more than 63 characters
|
|     with module.acm.module.registry.google_gke_hub_membership.primary[0]
|.    on .terraform/modules/acm/module/fleet-membership/membership.tf line 18 in resource "google_gke_hub_membership" "primary":
|.     18: resource "google_gke_hub_membership" "primary" {

NOTE
I am able to manually register this cluster on the Anthos page in the GCP console without issue.

### Expected behavior

Fleet registration succeeds.

### Observed behavior

Fleet registration fails.

### Terraform Configuration

```hcl
main.tf: 
module "acm" {
  source                     = "terraform-google-modules/kubernetes-engine/google//modules/acm"
  version                    = "23.0.0"
  project_id                 = var.project_id
  cluster_name               = var.cluster_name
  location                   = var.location
  sync_repo                  = var.acm_sync_repo
  sync_branch                = var.acm_sync_branch
  enable_fleet_feature       = var.enable_fleet_feature
  enable_policy_controller   = var.enable_policy_controller
  create_ssh_key             = var.create_ssh_key
  secret_type                = var.secret_type

provider.tf: 
provider "google" {
  project = var.project_id
  region  = var.region
}

provider "google-beta" {
  project = var.project_id
  region  = var.region
}

provider "kubernetes" {
  host     = "https://${var.cluster_endpoint}
  token    = data.google_client_config.default.access_token
  insecure = true
}

data.tf:
data "google_client_config" "default" {}

variables.tf: (has all of the above variables declared)

vars.tfvars:
project_id              = <some 17 character project_id>
region                  = us-central1
location                = us-central1
cluster_name            = <some long 36 character clustername>
cluster_endpoint        = <some.endpoint.ip.address>
enable_fleet_feature    = <true/false if first cluster in project>
acm_sync_repo           = <"https://mygithost.com/acm.git>

(git user/token being provided via gitlab vars and provided in separate resource block)
```

### Terraform Version

```sh
Terrform v1.2.1
```

**Comment by bharathkkb** [comment]
fixed by https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1429
